File: products/LoanHub/apps/backend/database/schemas/system_error.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class SystemErrorRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    request_id: str | None = None
    fingerprint: str
    user_id: UUID | None = None
    company_id: UUID | None = None
    branch_id: UUID | None = None
    method: str | None = None
    path: str
    status_code: int
    error_type: str
    message: str
    stack_trace: str | None = None
    severity: str
    environment: str | None = None
    user_agent: str | None = None
    context: dict[str, Any] = Field(default_factory=dict)
    occurrence_count: int
    first_seen_at: datetime
    last_seen_at: datetime
    is_resolved: bool
    resolved_at: datetime | None = None
    resolved_by_user_id: UUID | None = None
    resolution_notes: str | None = None
    created_at: datetime
    updated_at: datetime


class SystemErrorListRead(BaseModel):
    items: list[SystemErrorRead]
    total: int
    unresolved_count: int
    page: int
    page_size: int


class SystemErrorResolveCreate(BaseModel):
    resolution_notes: str = Field(min_length=2, max_length=4000)

File: products/LoanHub/apps/frontend/types/systemError.ts
export type SystemErrorLog = {
    id: string;
    request_id: string | null;
    fingerprint: string;
    user_id: string | null;
    company_id: string | null;
    branch_id: string | null;
    method: string | null;
    path: string;
    status_code: number;
    error_type: string;
    message: string;
    stack_trace: string | null;
    severity: string;
    environment: string | null;
    user_agent: string | null;
    context: Record<string, unknown>;
    occurrence_count: number;
    first_seen_at: string;
    last_seen_at: string;
    is_resolved: boolean;
    resolved_at: string | null;
    resolved_by_user_id: string | null;
    resolution_notes: string | null;
    created_at: string;
    updated_at: string;
};

export type SystemErrorListResponse = {
    items: SystemErrorLog[];
    total: number;
    unresolved_count: number;
    page: number;
    page_size: number;
};
